starport/pkg/xrelayer/config.go

```diff
@@ -35,23 +26,66 @@ var (
 	}
 )
 
+// confHome returns the home path of relayer
+func confHome() (string, error) {
```

Comment:
introducing confHome, confYamlPath, confFile funcs looks a bit too much just for dealing with joining file paths. I got an idea about how to keep one liners instead. so, how about the following?

We create a new pkg/ called xfilepath with an implementation like this:

```go
package xfilepath

type PathRetriever func(path string, err error)

func Path(path string) PathRetriever {
    return func() (string, error) { return path, nil }

func PathError(path string, err error) PathRetriever {
    return func() (string, error) { return path, err }
}

func Join(paths ...PathRetriever) PathRetriever // return on first non-nil err
```

And it can be used like this:
```go
var confHome = xfilepath.Join(xfilepath.PathError(os.UserHomeDir()), xfilepath.Path(".relayer"))
var confYamlPath = xfilepath.Join(xfilepath.PathError(confHome), xfilepath.Path"config/config.yaml"))

path, err := confhome()
path, err := confYamlPath()
```

Reply:
Yes, that a nice refactor. I implemented it for the whole PR